pnet_base/src/macaddr.rs

```diff
@@ -13,15 +13,82 @@ use std::str::FromStr;
 #[cfg(feature = "serde")]
 use serde::{de, Deserialize, Deserializer, Serialize, Serializer};
 
+/// The number of bytes in an ethernet (MAC) address.
+pub const ETHER_ADDR_LEN: usize = 6;
+
+/// Structure of a 48-bit Ethernet address.
+pub type EtherAddr = [u8; ETHER_ADDR_LEN];
+
+const LOCAL_ADDR_BIT: u8 = 0x02;
+const MULTICAST_ADDR_BIT: u8 = 0x01;
+
 /// A MAC address
-#[derive(PartialEq, Eq, Clone, Copy, Hash, Ord, PartialOrd)]
+#[derive(PartialEq, Eq, Clone, Copy, Default, Hash, Ord, PartialOrd)]
 pub struct MacAddr(pub u8, pub u8, pub u8, pub u8, pub u8, pub u8);
 
 impl MacAddr {
     /// Construct a new MacAddr
     pub fn new(a: u8, b: u8, c: u8, d: u8, e: u8, f: u8) -> MacAddr {
         MacAddr(a, b, c, d, e, f)
     }
+
+    /// Construct a empty MacAddr
+    pub fn empty() -> MacAddr {
```

Comment:
I don't think `empty`/`is_empty` make sense for addresses - it would be better to use `Option<MacAddr>` if you genuinely want an "empty" address.

Reply:
Consider the user case, an ARP probe is an ARP request constructed with an all-zero sender IP address (SPA), it's not a `Option<MacAddr>`.

I think we need another name instead of `empty`, maybe `pub fn zero() -> MacAddr` and `pub fn is_zero() -> bool` will be better?